src/main/java/hudson/plugins/xshell/XShellBuilder.java

```diff
@@ -115,9 +125,12 @@ public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
     EnvVars env = build.getEnvironment(listener);
     env.putAll(build.getBuildVariables());
 
+    // Determine the correct working directory
+    String absWorkingDir = build.getWorkspace() + (launcher.isUnix() ? UNIX_SEP : WINDOWS_SEP) + workingDir;

```

Comment:
Yes, I could modify this to accept an absolute path. `File.isAbsolute()` could be used to determine if the path is absolute. If so, it would be used as-is. Otherwise, it would be assumed to be relative to the workspace.